src/opnsense/mvc/app/controllers/OPNsense/BootEnvironments/Api/GeneralController.php

```diff
@@ -0,0 +1,200 @@
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+namespace OPNsense\BootEnvironments\Api;
+
+use OPNsense\Base\ApiMutableModelControllerBase;
+use OPNsense\Base\UIModelGrid;
+use OPNsense\Core\Backend;
+use OPNsense\Core\Config;
+use OPNsense\BootEnvironments\BootEnvironments;
+class GeneralController extends ApiMutableModelControllerBase
+{
+    protected static $internalModelName = 'general';
+    protected static $internalModelClass = '\OPNsense\BootEnvironments\General';
+
+    public function getAction()
+    {
+        // Check if we're only searching for a single record
+        $uri = explode("?", $_SERVER["REQUEST_URI"])[0];
+        $apiPath = '/api/bootenvironments/general/get/';
+
+        // request for new record?
+        if ($apiPath === $uri) {
+            $today = date("YmdHis");
+            return ['name' => 'BE'.$today];
+        } else if (substr($uri, 0, strlen($apiPath)) === $apiPath) {
+            // request for specific record?
+            $params = ltrim(str_replace($apiPath, '', $uri), '/');
+        }
+
+        $backEnd = new BackEnd();
+        $backResult = json_decode(trim($backEnd->configdRun('bootenvironments list')), true);
+        // only return valid json
+        if ($backResult !== null) {
+            // search for single record?
+            if (! empty($params)) {
+                foreach ($backResult as $result) {
+                    if ($result['uuid'] === $params) {
+                        return $result;
+                    }
+                }
+                return ['message' => 'Boot environment not found'];
+            }
+
+            // sort in order of creation
+            usort($backResult, function ($a, $b) use ($backResult) {
+                return strtotime($a['created']) - strtotime($b['created']);
+            });
+
+            // return all
+            return [
+                'status' => 'ok',
+                'path' => $params,
+                'rows' => $backResult,
+                'current' => 1,
+                'total' => count($backResult),
+                'rowCount' => 7,
+                'searchPhrase' => ''
+            ];
+        }
+        return ['message' => 'Unable to list boot environments'];
+    }
```

Comment:
```suggestion
    }
```

test

Reply:
@sheridans nope, won't let me add this to the PR. there is a setting somewhere at the right to enable maintainer edits

Reply:
![Screenshot_20240726_125431_Firefox](https://github.com/user-attachments/assets/16164a66-21c8-49cd-ab12-9ae489373b91)

It's ticked?

Reply:
Ok, appears to work then. It won’t let me commit empty changes then. Sorry for the noise.